Ihor/Netflix/src/MovieDetails/MovieDetailsMain/MovieDetailsMain.tsx

```diff
@@ -17,51 +19,58 @@ type MovieDetailsMainProps = IMovies &
   RouteComponentProps<{ id: string }> &
   PropsFromRedux;
 
-class MovieDetailsMain extends React.Component<MovieDetailsMainProps> {
-  componentDidMount() {
-    this.props.selectedMovieRequested(this.props.match.params.id);
-  }
 
-  componentDidUpdate(prevProps: MovieDetailsMainProps) {
-    if (this.props.movie !== prevProps.movie) {
-      const search = this.props.movie.genres[0];
-      this.props.moviesRequested("release_date", 2, "genre", search);
-    }
-    if (this.props.match.params.id !== prevProps.match.params.id) {
-      this.props.selectedMovieRequested(this.props.match.params.id);
-    }
-  }
+const MovieDetailsMain = (props: MovieDetailsMainProps) => {
+  useEffect(() => {
+    props.selectedMovieRequested({ id: props.match.params.id });
+  }, []);
+
+  useEffect(() => {
+    props.selectedMovieRequested({ id: props.match.params.id });
+  }, [props.match.params.id]);
 
-  increaseOffset = () => {
-    const search = this.props.movie.genres[0];
-    console.log(search)
-    this.props.loadData(this.props.offset + 10,"release_date","genre",search);
+  useEffect(() => {
+    const search = movie ? props.movie.genres[0] : "";
```

Comment:
You need to wrap your code in IF statement. We need to fetch movies only when we had the MOVIE, because our movies are based on the movie's genre